azure-ts-dynamicresource/index.ts

```diff
@@ -0,0 +1,86 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import { CDNCustomDomainResource } from "./cdnCustomDomain";
+
+const location = "West US";
+
+/** The custom domain host of the CDN endpoint. */
+export let cdnCustomDomainResource: CDNCustomDomainResource | undefined;
+
+// Create an Azure Resource Group
+const resourceGroup = new azure.core.ResourceGroup("resourceGroup", {
+    location: location,
+});
+
+// Create an Azure resource (Storage Account) so we can point the CDN endpoint to it.
+const storageAccount = new azure.storage.Account("storage-account", {
+    resourceGroupName: resourceGroup.name,
+    location: resourceGroup.location,
+    accountTier: "Standard",
+    accountReplicationType: "LRS",
+});
+
+// Create a Blob container in the storage account.
+const blobContainer = new azure.storage.Container("blob-container", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    containerAccessType: "blob"
+});
+
+const cdnProfile = new azure.cdn.Profile("cdn-profile", {
+    location: location,
+    resourceGroupName: resourceGroup.name,
+    sku: "Standard_Akamai",
+});
+
+const cdnEndpoint = new azure.cdn.Endpoint("my-cdn-endpoint", {
+    name: "my-cdn-endpoint",
+    resourceGroupName: resourceGroup.name,
+    profileName: cdnProfile.name,
+    isHttpsAllowed: true,
+    isHttpAllowed: false,
+    location: location,
+    isCompressionEnabled: true,
+    originHostHeader: storageAccount.primaryBlobEndpoint,
+    contentTypesToCompresses: [
+        "text/plain",
+        "text/html",
+        "text/css",
+        "text/javascript",
+        "application/x-javascript",
+        "application/javascript",
+        "application/json",
+        "application/xml",
+        "image/png",
+        "image/jpeg"
+    ],
+    origins: [
+        {
+            name: "cdn-origin",
+            hostName: storageAccount.primaryBlobEndpoint,
```

Comment:
This needs to be a hostname like `bla.blob.core.windows.net` not the full URL like `https://bla.blob.core.windows.net`. Consider using something like `storageAccount.primaryBlobEndpoint.apply(ep => url.parse(ep).hostname!)`. Applies to `originHostHeader` too.

Reply:
`primaryBlobEndpoint` actually does work for both of those properties. But if you think mapping the hostname would be better, I can change it.

Reply:
Hmm, I'm getting errors running that... weird

Reply:
Ah sorry. I compared this with the wrong project of mine. In my personal project, I am using the function app's host name as the origin. When I translated this to use the storage account, I used the wrong property. I should have used `primaryBlobHost`.